connect_ext_ppr/webapp.py

```diff
@@ -25,13 +25,34 @@
     _get_extension_client,
     _get_installation,
     filter_products_list,
+    get_deployment_schema,
     get_products,
 )
 
 
 @web_app(router)
 class ConnectExtensionXvsWebApplication(WebApplicationBase):
 
+    @router.get(
+        '/deployments/{deployment_id}',
+        summary='Deployment details',
+        response_model=DeploymentSchema,
+    )
+    def get_deployment(
+        self,
+        deployment_id: str,
+        client: ConnectClient = Depends(get_installation_client),
+        db: VerboseBaseSession = Depends(get_db),
+    ):
+        dep = db.query(Deployment).filter(Deployment.id == deployment_id).first()
```

Comment:
I found that if we do not filter by account, we are going to retrieve a Deployment that should not be visible for the account that is making the request, so I will ask for some refactor:

connect_ext_ppr/models/deployment.py -> Remove `unique=True` constraint from `product_id` field of `Deployment` model.
connect_ext_ppr/models/utils.py -> Add `account_id=installation['owner']['id']` to filter query. Line: 9
connect_ext_ppr/events.py - > Add `account_id=installation['owner']['id']` to filter query. Note: installation in events is available as instance attr `self.installation`. Lines: 43, 82
connect_ext_ppr/webapp.py  Add `installation: dict = Depends(get_installation),` to `get_deployments` route, and add `account_id=installation['owner']['id']` to filter query. Lines: 46
You can also use `filter_by` that allow you to specify kwargs as Django filter: `db.query(Deployment).filter_by(account_id=self.installation['owner']['id']`, product=request['product']['id'])`

Reply:
Fixed